util/reggen/register.py

```diff
@@ -431,6 +445,22 @@ def check_valid_regwen(self) -> None:
                                  wen_fld.swaccess.key,
                                  wen_fld.hwaccess.key))
 
+    def handle_async(self, clocks: Clocking) -> object:
```

Comment:
I think this returns `Optional[ClockingItem]`?

Reply:
ah i originally had that, but i got this during lint 

```register.py:460: error: Incompatible return value type (got "object", expected "Optional[ClockingItem]")```

Reply:
Yep, that's because `Clocking.get_by_clock` was returning an `object` rather than an `Optional[ClockingItem]`.